src/test/PoolModifyLiquidityTest.sol

```diff
@@ -72,10 +74,10 @@ contract PoolModifyLiquidityTest is PoolTestBase {
             assert(!(delta0 > 0 || delta1 > 0));
         }
 
-        if (delta0 < 0) _settle(data.key.currency0, data.sender, int128(delta0), data.settleUsingTransfer);
-        if (delta1 < 0) _settle(data.key.currency1, data.sender, int128(delta1), data.settleUsingTransfer);
-        if (delta0 > 0) _take(data.key.currency0, data.sender, int128(delta0), data.withdrawTokens);
-        if (delta1 > 0) _take(data.key.currency1, data.sender, int128(delta1), data.withdrawTokens);
+        if (delta0 < 0) data.key.currency0.settle(manager, data.sender, uint256(-delta0), !data.settleUsingTransfer);
```

Comment:
I'd just change the testSettings params to match the underlying burn/claims bool... ie 

```
    struct TestSettings {
        bool takeClaims; // instead of withdrawTokens
        bool settleWithBurn; // instead of settleUsingTransfer
        bool currencyAlreadySent;
    }
```

that way here you can just pass in data.settleWithBurn/data.takeClaims